### PR TITLE
feat(go): cloud run env test

### PR DESCRIPTION
**Pull request by nicoleczhu**
envctl commands work. 

Assuming common test names: 
- simplelog
- standardlog

### CODE REVIEW

File: tests/go/test_appengine_flex_python.py [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+
+import google.cloud.logging
+
+from ..common.common import Common
+
+
+class TestAppEngineFlex(Common, unittest.TestCase):
+
+    environment = "appengine_flex_python"
+    language = "go"
```
Comment (Daniel-Sanche) [inline tests/go/test_appengine_flex_python.py:26]: Do you need this file? In Python I made two different test environments for the Python runtime and the container runtime. I'm not sure if Go is the same situation. I don't think we need an exact match of all of these test files though

File: deployable/go/main.go [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+		}
+		log.Printf("Listening on port %s", port)
+		if err := http.ListenAndServe(":"+port, nil); err != nil {
+			log.Fatal(err)
+		}
+	}
+}
+
+// [Optional] envctl go <env> trigger simpleLog logname=foo logtext=bar
+func simpleLog(args map[string]string) {
```
[the remark this answers is not in the document]
Comment (nicoleczhu) [reply deployable/go/main.go:87]: ah yes I noticed. I'll refactor this later when we add additional tests. Handy to have it all in one file for me atm :)

File: tests/go/test_appengine_flex_python.py (hunk 1 above)
Comment (nicoleczhu) [reply tests/go/test_appengine_flex_python.py:26]: yes . will remove thank you

File: README.md [hunk 3]
```diff
@@ -64,3 +65,8 @@ Implementation code for each language/environment pair can be found in `envctl/e
 Contains the pytest code that tests each environment. Common logic is stored as abstract super-classes in `tests/common`.
 Concrete implementations for each environment can be found in `tests/<language>/test_<environment>.py`. 
 Test files in `tests/<language>` can inherit from any file in `tests/common` logic as needed, in order to share test logic between environments.
+
+Current common tests include:
+
+- `simplelog(logname, logtext)`: tests logging a simple text log
+- `standardlog()`: tests logging with standard library wrapper
```
[the remark this answers is not in the document]
Comment (nicoleczhu) [reply README.md:72]: yes will add

File: envctl/env_scripts/go/cloudrun.sh [hunk 4]
```diff
@@ -0,0 +1,106 @@
...
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+set -e # exit on any failure
+set -o pipefail # any step in pipe caused failure
+set -u # undefined variables cause exit
+set -x
```
Comment (Daniel-Sanche) [inline envctl/env_scripts/go/cloudrun.sh:19]: nit: I find this a bit noisy for regular use, I'd prefer to turn it on and off when needed. But up to you

Comment (nicoleczhu) [reply envctl/env_scripts/go/cloudrun.sh]: ah yes removed

File: deployable/go/main.go [hunk 5]
```diff
@@ -0,0 +1,111 @@
...
+		http.Error(w, "Bad Request", http.StatusBadRequest)
+		return
+	}
+
+	msg := string(m.Message.Data)
+	args := m.Message.Attributes
+
+	switch msg {
+	case "simplelog":
+		simpleLog(args)
```
Comment (Daniel-Sanche) [inline deployable/go/main.go:60]: It kind of threw me off that the function is `simpleLog` but it is triggered with `simplelog`. Can we make them match?

File: deployable/go/main.go [hunk 6]
```diff
@@ -0,0 +1,111 @@
...
+			log.Printf("Defaulting to port %s", port)
+		}
+		log.Printf("Listening on port %s", port)
+		if err := http.ListenAndServe(":"+port, nil); err != nil {
+			log.Fatal(err)
+		}
+	}
+}
+
+// [Optional] envctl go <env> trigger simpleLog logname=foo logtext=bar
```
Comment (Daniel-Sanche) [inline deployable/go/main.go:86]: I think the proper format is actually `envctl go <env> trigger simpleLog logname=foo,logtext=bar`. The pubsub attributes are all a single argument to envctl

**Review by Daniel-Sanche — APPROVED** [review]
A couple more small comments, but other than that LGTM

File: deployable/go/main.go (hunk 5 above)
Comment (nicoleczhu) [reply deployable/go/main.go]: fixed

File: deployable/go/main.go (hunk 6 above)
Comment (nicoleczhu) [reply deployable/go/main.go]: fixed

**Pull request merged**